src/python/build_management/build_manager.py

```diff
@@ -1250,7 +1250,11 @@ def setup_regular_build(revision,
   base_build_dir = _base_build_dir(bucket_path)
 
   build_class = RegularBuild
-  if environment.is_trusted_host():
+
+  # Testcases are run on the trusted host when launcher is used. Otherwise,
+  # set up the build on the untrusted worker bot.
+  launcher = environment.get_value('LAUNCHER_PATH')
+  if environment.is_trusted_host() and not launcher:
```

Comment:
Did something change here, or was this always broken?

We shouldn't ever run builds on the trusted host if the builds are untrusted. Resources are *very* limited there. 

Also we shouldn't add a dependency on the fuzzer being setup before this code is run here.